### PR TITLE
fixes #8516 - removing the require 'fog' from the project

### PR DESCRIPTION
**Pull request by unorthodoxgeek**
requiring only what we use

### CODE REVIEW

File: app/models/concerns/fog_extensions.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ module FogExtensions
 end
 
 begin
-  require 'fog'
+  #require 'fog'
```
Comment (ohadlevy) [inline app/models/concerns/fog_extensions.rb]: hmm.. where does this gets required then?
and a nitpick, please remove the line vs commenting it out.

Comment (unorthodoxgeek) [inline app/models/concerns/fog_extensions.rb]: the point is that we don't want it required, but we require the modules we're using (in the next lines in this file). i.e. fog/aws, fog/rackspace or whatever else.

removing the line, commented it out for testing, forgot to remove...

File: app/models/concerns/fog_extensions.rb [hunk 2]
```diff
@@ -2,7 +2,6 @@ module FogExtensions
 end
 
 begin
-  require 'fog'
```
Comment (ohadlevy) [inline app/models/concerns/fog_extensions.rb:5]: AFAIR we kept it because the next line mp fog::model which is not for a specific provider.

Comment (unorthodoxgeek) [inline app/models/concerns/fog_extensions.rb:5]: it's part of the fog-core which is required by bundler.

**Comment by domcleal** [comment]
https://github.com/theforeman/foreman/blob/develop/app/services/foreman/provision/ssh.rb#L2 too.

**Comment by unorthodoxgeek** [comment]
yup. also, the requirement of fog there is futile, as Fog::SSH and Fog::SCP are part of fog-core as well...

**Comment by ohadlevy** [comment]
when testing with the GCE CR i got:

```
Error has occurred while listing VMs on GCE-ohad (europe-west1-b-Google): uninitialized constant Fog::VERSION
/home/ohad/.rvm/gems/ruby-2.1.5@foreman/gems/fog-1.25.0/lib/fog/google/core.rb:107:in `new_pk12_google_client'
/home/ohad/.rvm/gems/ruby-2.1.5@foreman/gems/fog-1.25.0/lib/fog/google/core.rb:85:in `initialize_google_client'
/home/ohad/.rvm/gems/ruby-2.1.5@foreman/gems/fog-1.25.0/lib/fog/google/compute.rb:1058:in `initialize'
/home/ohad/.rvm/gems/ruby-2.1.5@foreman/gems/fog-core-1.25.0/lib/fog/core/service.rb:115:in `new'
/home/ohad/.rvm/gems/ruby-2.1.5@foreman/gems/fog-core-1.25.0/lib/fog/core/service.rb:115:in `new'
/home/ohad/.rvm/gems/ruby-2.1.5@foreman/gems/fog-core-1.25.0/lib/fog/compute.rb:60:in `new'
```

**Comment by domcleal** [comment]
There were some bugs around Fog::VERSION in 1.25.0 that I think were fixed in 1.26, as it was moving around between fog/fog-core and fog/version wasn't required.

Ah, here it is: https://github.com/fog/fog/issues/3281

**Comment by domcleal** [comment]
@unorthodoxgeek could you rebase this please?  Hopefully it'll resolve the error that @ohadlevy experienced.

**Comment by dLobatog** [comment]
@unorthodoxgeek ping?

**Comment by unorthodoxgeek** [comment]
pong

**Comment by dLobatog** [comment]
Merged as a180c3213ff9bd26eb17c3c16cc8c6412afc323f, thanks @unorthodoxgeek !

**Pull request closed without merging**